source/npm/util.js

```diff
@@ -109,3 +111,57 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromFileAttribute) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromFileAttribute) {
+		result = getFilesNotPartOfFileAttribute(filesFromFileAttribute, newFiles);
```

Comment:
See this: https://docs.npmjs.com/files/package.json#files

There are more things to consider here, such as:

> You can also provide a `.npmignore` file in the root of your package or in subdirectories, which will keep files from being included. At the root of your package it will not override the “files” field, but in subdirectories it will. The .npmignore file works just like a `.gitignore`. If there is a .gitignore file, and `.npmignore` is missing, `.gitignore`’s contents will be used instead.

Reply:
The `.npmignore`-files in the subdirectories is one feature i didn't implemented yet.
The priority (file-attribute in `package.json` vs. `.npmignore` in the root of the package i test again)
The `.gitignore`-file i think we shouldn't consider in this pull-request.
This pull-request just checks files, which were added to the version-control-system git
and are not published in the released package.
Files, which are ignored by `.gitignore` should be checked in other algorithm, because they are just local files on the developer machine and cannot be checked in to the version-control-system. They can be just part of the released package. It would only make sense to show a warning, if the released packages contains local files.